File: archive/g/go/bubble-sort.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"regexp"
	"strconv"
	"strings"
)

func bubbleSort(list []int) []int {
	swapped := true
	for swapped {
		swapped = false
		for i := 0; i < len(list)-1; i++ {
			if list[i] > list[i+1] {
				n := list[i]
				list[i] = list[i+1]
				list[i+1] = n
				swapped = true
			}
		}
	}
	return list
}

func swap(list []int, firstIndex int, secondIndex int) bool {
	if list[firstIndex] > list[secondIndex] {
		x := list[firstIndex]
		list[firstIndex] = list[secondIndex]
		list[secondIndex] = x
		return true
	}
	return false
}

func exitWithError() {
	fmt.Println("Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"")
	os.Exit(1)
}

func main() {
	if len(os.Args) != 2 {
		exitWithError()
	}

	sNums := regexp.MustCompile(", ?").Split(os.Args[1], -1)
	if len(sNums) <= 2 {
		exitWithError()
	}

	var nums []int
	for _, num := range sNums {
		n, err := strconv.Atoi(num)
		if err != nil {
			exitWithError()
		}
		nums = append(nums, n)
	}
	nums = bubbleSort(nums)
	str, _ := json.Marshal(nums)

	fmt.Println(strings.Trim(string(str), "[]"))
}

File: archive/k/kotlin/BubbleSort.kt
fun main(args: Array<String>) 
{
    var nums: IntArray
    try
    {
        nums = args[0].split(", ").map{ it.toInt() }.toIntArray()
        if (nums.size < 2) {
            throw Exception()
        }
    }
    catch(e: Exception)
    {
        println("Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"")
        return
    }
    var swapped:Boolean = false
    for(i in 0 until nums.count()-1)
    {
        swapped = false
        for(j in 0 until nums.count()-i-1)
        {
            if(nums[j]>nums[j+1])
            {
                //swap
                nums[j] = nums[j+1].also {nums[j+1] =  nums[j]}
                swapped = true
            }
        }
        if (swapped==false)
        {
            break
        }
    }
    for(i in 0 until nums.count())
    {
        if (i==nums.count()-1)
        {
            println("${nums[i]}")
            return
        }
        print("${nums[i]}, ")
    }
}
